docs: Updates installation instruction for Debian/Ubuntu (#64232)
grafana/grafana: docs/sources/setup-grafana/installation/debian/index.md

# Install on Debian or Ubuntu

This topic explains how to install Grafana dependencies, install Grafana on Linux Debian or Ubuntu, get the service up and running on your Debian or Ubuntu system, and describes the contents of the installation package.

> You can install Grafana using the Grafana Labs APT repository, by downloading a `.deb` package, or by downloading a binary `.tar.gz` file.

Reviewer: ```suggestion
You can install Grafana using the Grafana Labs APT repository, by downloading a `.deb` package, or by downloading a binary `.tar.gz` file.

If you install via the `.deb` package or `.tar.gz` file, then you will need to manually update Grafana for each new version.
```
We might want to keep the APT repo at the top here as that is the preferred install mechanism, and ideally we want to make that more prominent on the download page as well.

Author: ok, cool. I'll make that section up.

## Download and install Grafana

Complete the following steps to download and install Grafana on Debian or Ubuntu:

1. Navigate to the [Grafana download page](https://grafana.com/grafana/download).
1. Select the Grafana version you want to install.
   - The most recent Grafana version is selected by default.
   - The **Version** field displays only tagged releases. If you want to install a nightly build, click **Nightly Builds** and then select a version.
1. Select an **Edition**.
   - **Enterprise:** This is the recommended version. It is functionally identical to the open source version, but includes features you can unlock with a license, if you so choose.
   - **Open Source:** This version is functionally identical to the Enterprise version, but you will need to download the Enterprise version if you want Enterprise features.
1. Depending on which system you are running, click the **Linux** or **ARM** tab on the download page.
1. Copy and paste the code from the installation page into your command line and run.

> ### Install from APT repository

Author: @DanCech , I'm assuming that this is the section that we should keep in? Or was it only part of this section?

Reviewer: Yup, we want to keep the apt instructions

Author: Great, thanks.

This page explains how to install Grafana dependencies, download and install Grafana, get the service up and running on your Debian or Ubuntu system, and also describes the installation package details.

## Note on upgrading

While the process for upgrading Grafana is very similar to installing Grafana, there are some key backup steps you should perform. Read [Upgrading Grafana]({{< relref "../../../upgrade-guide/" >}}) for tips and guidance on updating an existing installation.

## Download and install Grafana

Complete the following steps to download and install Grafana:

1. On the [Grafana download page](https://grafana.com/grafana/download), select the Grafana version you want to install.
   - The most recent Grafana version is selected by default.

>    - The **Version** field displays only finished releases. If you want to install a beta version, click **Nightly Builds** and then select a version.

Reviewer: Beta releases are still tagged and will show up in the version select dropdown (though the most recent stable release will still be the default), the user only needs to click the Nightly Builds button if they want to run a nightly build (though I don't think many people actually do that, will dig into the usage reports).
```suggestion
   - The **Version** field displays only tagged releases. If you want to install a nightly build, click **Nightly Builds** and then select a version.
```

1. Select an **Edition**.
   - **Enterprise:** This is the recommended version. It is functionally identical to the open source version, but includes features you can unlock with a license if you so choose.
   - **Open Source:** This version is functionally identical to the Enterprise version, but you will need to download the Enterprise version if you want Enterprise features.
1. Depending on which system you are running, click the **Linux** or **ARM** tab on the download page.
1. Copy and paste the code from the installation page into your command line and run.

> ---

Author: @DanCech, from here forward through line 108, I'm unsure what, if any, of these instructions we should keep (they are very old). I want this topic to point to the download page where they select the release and copy and paste the instructions into the command line.

Reviewer: We will need to retain the apt repo setup instructions, and in some ways it makes more sense for those to be here rather than on the download page

Author: Ok, good to know and I agree that we should keep them in the markdown. Thanks!

> **Note:** Grafana Enterprise is the recommended and default edition. It is available for free and includes all the features of the OSS edition. You can also upgrade to the [full Enterprise feature set](https://grafana.com/products/enterprise/?utm_source=grafana-install-page), which has support for [Enterprise plugins](https://grafana.com/grafana/plugins/?enterprise=1&utcm_source=grafana-install-page).

> ### Before you begin

Author: @DanCech , this section is meant to provide a list of prerequisites, for example, the user should have certain permissions in place before they start the task steps. Can you think of any such prereqs? If we don't have any, we can remove this section.

Author: will removing

To follow these instructions, you need:

- xxx
- xxx

### Steps

Complete the following steps to install Grafana from the APT repository:

> 1. To install the latest release, run the following commands:

Author: @DanCech , each of the other steps in this section begin with "To add a repository...". Is it the same case here? Should this step be "To add a latest release repository, run..."?

Reviewer: ```suggestion
1. To install required packages and download the Grafana repository signing key, run the following commands:
```

   ```bash
   sudo apt-get install -y apt-transport-https
   sudo apt-get install -y software-properties-common wget
   sudo wget -q -O /usr/share/keyrings/grafana.key https://apt.grafana.com/gpg.key
   ```

1. To add a repository for stable releases, run the following commands:

   ```bash
   echo "deb [signed-by=/usr/share/keyrings/grafana.key] https://apt.grafana.com stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list
   ```

1. To add a respository for beta releases, run the following commands:

   ```bash
   echo "deb [signed-by=/usr/share/keyrings/grafana.key] https://apt.grafana.com beta main" | sudo tee -a /etc/apt/sources.list.d/grafana.list
   ```

> 1. After you add the repository, run one of the following commands:

Reviewer: ```suggestion
1. After you add the repository, run the following commands to install the OSS or Enterprise release:
```

>    ```bash
>    sudo apt-get update

Reviewer: ```suggestion
   ```bash
   # update the list of available packages
   sudo apt-get update
```

   # Install the latest OSS release:
   sudo apt-get install grafana

   # Install the latest Enterprise release:
   sudo apt-get install grafana-enterprise
   ```

> ## Install Grafana using DEB or the standalone binaries

Reviewer: ```suggestion
## Install Grafana using deb package or the standalone binaries
```

> If you choose not to install Grafana using APT, you can download and install Grafana using DEB or the standalone binaries.

Reviewer: ```suggestion
If you choose not to install Grafana using APT, you can download and install Grafana using the deb package or as a standalone binary.
```

> ### Before you begin

Author: @DanCech let's add prereqs if we have 'em.

Reviewer: prereqs are already included in the download instructions

Author: will remove.

To follow these instructions, you need:

- xxx
- xxx

### Steps

Complete the following steps to install Grafana using DEB or the standalone binaries:

1. Navigate to the [Grafana download page](https://grafana.com/grafana/download).
1. Select the Grafana version you want to install.
   - The most recent Grafana version is selected by default.
   - The **Version** field displays only tagged releases. If you want to install a nightly build, click **Nightly Builds** and then select a version.
1. Select an **Edition**.
   - **Enterprise:** This is the recommended version. It is functionally identical to the open source version, but includes features you can unlock with a license, if you so choose.
   - **Open Source:** This version is functionally identical to the Enterprise version, but you will need to download the Enterprise version if you want Enterprise features.
1. Depending on which system you are running, click the **Linux** or **ARM** tab on the download page.
1. Copy and paste the code from the installation page into your command line and run.

> ## 2. Start the server

Author: @DanCech , I know this looks strange, but need to keep as is or the link from the download page breaks. Post merge cleanup work.

The following sections provide instructions for starting the `grafana-server` process as the `grafana` user, which was created during the package installation.

If you installed with the APT repository or `.deb` package, then you can start the server using `systemd` or `init.d`. If you installed a binary `.tar.gz` file, then you need to execute the binary.

### Start the Grafana server with systemd

Complete the following steps to start the Grafana server with systemd and verify that it is running:

1. To start the service, run the following commands:

   ```bash
   sudo systemctl daemon-reload
   sudo systemctl start grafana-server
   sudo systemctl status grafana-server
   ```

> 1. To verify that the service is running, run the following command:

Author: @DanCech , the original content makes mention of verifying that the server is up and running, which I think is a good idea. But, there were no instructions for how to do that. Would you know what to put here?

Reviewer: `sudo systemctl status grafana-server` is the command that will tell the user whether the service is running

   ```

>    Need input here

Author: ```suggestion
   sudo systemctl status grafana-server
```

   ```

1. To configure the Grafana server to start at boot, run the following command:

   ```bash
   sudo systemctl enable grafana-server.service
   ```

#### Serve Grafana on a port < 1024

{{< docs/shared "systemd/bind-net-capabilities.md" >}}

### Start the server with init.d

Complete the following steps to start the Grafana service and verify that it is running:

1. To start the Grafana server, run the following commands:

   ```bash
   sudo service grafana-server start
   sudo service grafana-server status
   ```

> 1. To verify that the service is running, run the following command:

Author: @DanCech , same here, need a command, please.

Reviewer: same here `sudo service grafana-server status`

   ```

>    Need input here

Author: ```suggestion
   sudo service grafana-server status
```

sudo apt-get install grafana-enterprise
```

> ### Install .deb package

Author: @DanCech , removed all this content because I think we can rely on the download page for guidance. Does that work?

> ## Start the Grafana server

Author: @DanCech, I recommend moving all start server content out of the installation topic and into a topic previously known as "Restart the Grafana server", which I have hijacked and am now calling "Start the Grafana Server". This PR includes a first pass at moving the Linux, Mac, and Windows instructions there.

However, these instructions must remain in this topic for now because there are links from the download page to this content. If you agree with this approach, let's coordinate on changing those links in the download page after this PR is merged. Then I'll remove the start server instructions from this topic.

- The default configuration specifies a SQLite3 db at `/var/lib/grafana/grafana.db`

> - Installs HTML/JS/CSS and other Grafana files at `/usr/share/grafana`

Author: dd

Reviewer: probably not

Author: Great, will remove. thx

- The default configuration specifies a SQLite3 db at `/var/lib/grafana/grafana.db`
- Installs HTML/JS/CSS and other Grafana files at `/usr/share/grafana`

## Upgrade Grafana

While the process for upgrading Grafana is very similar to installing Grafana, there are important backup tasks you should perform. Refer to [Upgrade Grafana]({{< relref "../../../upgrade-guide/" >}}) for guidance on updating an existing installation.

## Next steps

> - [Start the Grafana server]({{< relref "../../../restart-grafana/" >}})

Reviewer: next steps should be to configure grafana

Author: keeping as is based on our discussion.
